modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorage.java

```diff
@@ -322,6 +376,74 @@ public long leaseStartTime() {
         });
     }
 
+    // TODO https://issues.apache.org/jira/browse/IGNITE-15119 nodeId type should be changed from String to UUID, after the fix
+    // TODO nodeID will be stored in meta directly and not the blob storage.
+    @Override
+    public @Nullable String primaryReplicaNodeId() {
+        return busy(() -> {
+            throwExceptionIfStorageNotInRunnableState();
+
+            try {
+                primaryReplicaMetaReadWriteLock.readLock().lock();
+
+                try {
```

Comment:
Why do we need the external `try`? Can't we move the `catch` logic to the internal one?

Reply:
Sure. Fixed.